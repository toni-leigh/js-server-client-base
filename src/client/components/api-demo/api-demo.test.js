import React from 'react';
import { createStore } from 'redux';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';

import {
  ApiDemo,
  ApiDemoConnected,
  mapDispatchToProps,
  mapStateToProps
} from './api-demo';

import rootReducer from '../../reducers/reducers';

describe('<ApiDemo />', () => {
  let apiDemo;

  beforeAll(() => {
    apiDemo = shallow(
      <ApiDemo
        getHardcodedText={ jest.fn() }
        hardcodedText='Test text'
      />
    );
  });

  test('basic dumb component render', () => {
    expect(apiDemo).toMatchSnapshot();
  });
});

describe('<ApiDemoConnected />', () => {
  let apiDemoConnected;

  beforeAll(() => {
    const store = createStore(rootReducer);
    apiDemoConnected = shallow(
      <ApiDemoConnected store={ store } />
    );
  });

  test('render occurs through connection', () => {
    expect(apiDemoConnected).toMatchSnapshot();
  });
});

describe('mapDispatchToProps', () => {
  test('processes the actions correctly', () => {
    expect(mapDispatchToProps).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  test('processes the state correctly', () => {
    const state = fromJS({ hardcodedText: 'Test Text' });
    expect(mapStateToProps(state)).toMatchSnapshot();
  });
});
