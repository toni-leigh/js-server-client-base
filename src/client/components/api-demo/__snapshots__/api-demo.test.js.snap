// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApiDemo /> basic dumb component render 1`] = `
<div
  className="server-client-api-demo"
>
  Test text
</div>
`;

exports[`<ApiDemoConnected /> render occurs through connection 1`] = `
<withImmutablePropsToJS(ApiDemo)
  getHardcodedText={[Function]}
  hardcodedText=""
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`mapDispatchToProps processes the actions correctly 1`] = `
Object {
  "getHardcodedText": [Function],
}
`;

exports[`mapStateToProps processes the state correctly 1`] = `
Object {
  "hardcodedText": "Test Text",
}
`;
